src/controllers/dashBoardController.spec.js

```diff
@@ -0,0 +1,698 @@
+      );
+    })
+
+    test('Returns 200 if leaderboard data is empty and returns getUserLaborData', async () => {
+      const dashboardHelperObject = {
+        getLeaderboard: jest.fn(() => Promise.resolve([])),
+        getUserLaborData: jest.fn(() => Promise.resolve([]))
+      };
+
+      dashboardHelperClosure.mockImplementationOnce(() => dashboardHelperObject);
+
+      const { leaderboarddata } = makeSut();
+
+      const response = leaderboarddata(mockReq, mockRes);
+
+      await flushPromises();
+
+      assertResMock(
+        200,
+        [],
+        response,
+        mockRes,
+      );
+    })
+
+    test('Returns 400 if there is an error', async () => {
+      const dashboardHelperObject = {
+        getLeaderboard: jest.fn(() => Promise.reject({}))
+      };
+
+      dashboardHelperClosure.mockImplementationOnce(() => dashboardHelperObject);
+
+      const { leaderboarddata } = makeSut();
+
+      const response = leaderboarddata(mockReq, mockRes);
+
+      await flushPromises();
+
+      assertResMock(
+        400,
+        {},
+        response,
+        mockRes,
+      );
+    })
+  })
+  
+  describe('orgData Tests', () => {
+    
+    test('Returns 400 if there is an error in the function', async () => {
+
+      const dashboardHelperObject = {
+        getOrgData: jest.fn(() => Promise.reject(error))
+      };
+
+      dashboardHelperClosure.mockImplementationOnce(() => dashboardHelperObject);
+
+      const { orgData } = makeSut();
+
+      const response = orgData(mockReq, mockRes);
+
+      await flushPromises();
+
+      assertResMock(
+        400,
+        error,
+        response,
+        mockRes,
+      );
+    })
+
+    test('Returns 200 if the result is found and returns result', async () => {
+      const mockResult = { id: 1, name: 'Mock Results'};
+
+      const dashboardHelperObject = {
+        getOrgData: jest.fn(() => Promise.resolve([mockResult]))
+      }
+
+      dashboardHelperClosure.mockImplementationOnce(() => dashboardHelperObject);
+
+      const { orgData } = makeSut();
+
+      const response = orgData(mockReq, mockRes);
+
+      await flushPromises();
+
+      assertResMock(
+        200,
+        mockResult,
+        response,
+        mockRes,
+      );
+    })
+  });
+
+  describe('dashboarddata Tests', () => {
+    test('Returns 200 if there is no error and return results', async () => {
+
+      const dashboardHelperObject = {
+        personaldetails: jest.fn(() => Promise.resolve({}))
+      }
+
+      dashboardHelperClosure.mockImplementationOnce(() => dashboardHelperObject);
+
+      const { dashboarddata } = makeSut();
+
+      const response = dashboarddata(mockReq, mockRes);
+
+      await flushPromises();
+
+      assertResMock(
+        200,
+        {},
+        response,
+        mockRes,
+      )
+    })
+
+  });
+  
+  describe('sendBugReport Tests', () => {
+
+    test('Returns 200 if the bug report email is sent ', async () => {
+      const mockReq = {
```

Comment:
Hey Lin, 
Instead of `const mockReq = { ... }`, could you try the following:

```
mockReq.body = { 
          ...mockReq.body, 
          firstName: 'Lin',
          lastName: 'Test',
          title: 'Bug in feature X',
          environment: 'macOS 10.15, Chrome 89, App version 1.2.3',
          reproduction: '1. Click on button A\n2. Enter valid data\n3. Click submit',
          expected: 'The app should not display an error message',
          actual: 'The app',
          visual: 'Screenshot attached',
          severity: 'High',
          email: 'lin.test@example.com',
};
```

I tried the above suggestion and the test run correctly.

`Image Reference:`
<img width="761" alt="Screenshot 2024-07-12 at 9 36 31 PM" src="https://github.com/user-attachments/assets/688cb008-28f3-4c54-82ab-07640ba91f96">

Reply:
There are few other lines as well where you are redeclaring the mockReq variable, it would be better to try the above suggestion there as well.